src/Connect.js

```diff
@@ -1,68 +1,551 @@
+  *  @param    {Object}       abi                                   contract ABI
+  *  @return   {Object}                                             contract object
   */
-  getWeb3 () {
-    const provider = this.getProvider()
-    const web3 = new Web3()
-    web3.setProvider(provider)
-    // Work around to issue with web3 requiring a from parameter. This isn't actually used.
-    web3.eth.defaultAccount = '0xB42E70a3c6dd57003f4bFe7B06E370d21CDA8087'
-    return web3
+  contract (abi) {
+    //TODO could have default id as method name instead of txReq?
+    const txObjHandler = (txObj, id) => {
+      txObj.fn = txObj.function
+      delete txObj['function']
+      return this.sendTransaction(txObj, id)
+    }
+    return ContractFactory(txObjHandler.bind(this))(abi)
+  }
+
+  /**
+   *  Given a transaction object (similarly defined as the web3 transaction object)
+   *  it creates a transaction sign request and sends it.
+   *
+   *  @example
+   *  const txobject = {
+   *    to: '0xc3245e75d3ecd1e81a9bfb6558b6dafe71e9f347',
+   *    value: '0.1',
+   *    fn: "setStatus(string 'hello', bytes32 '0xc3245e75d3ecd1e81a9bfb6558b6dafe71e9f347')",
+   *    appName: 'MyDapp'
+   *  }
+   *  connect.sendTransaction(txobject, 'setStatus')
+   *  connect.onResponse('setStatus').then(res => {
+   *    const txId = res.res
+   *  })
+   *
+   *  @param    {Object}    txObj
+   *  @param    {String}    [id='txReq']    string to identify request, later used to get response, by default name of function, if not function call, by default 'txReq'
+   */
+   sendTransaction (txObj, id) {
+     txObj.to = isMNID(txObj.to) ? txObj.to : encode({network: this.network.id, address: txObj.to})
+     //  Create default id, where id is function name, or txReq if no function name
+     if (!id) id = txObj.fn ? txObj.fn.split('(')[0] : 'txReq'
+     this.credentials.txRequest(txObj, {callbackUrl: this.genCallback(id)})
+                     .then(jwt => this.send(jwt, id))
+   }
+
+  //  TODO this name is confusing
+  /**
+   *  Request uPort client/user to sign a claim or list of claims
+   *
+   *  @example
+   *  const unsignedClaim = {
+   *    claim: {
+   *      "Citizen of city X": {
+   *        "Allowed to vote": true,
+   *        "Document": "QmZZBBKPS2NWc6PMZbUk9zUHCo1SHKzQPPX4ndfwaYzmPW"
+   *      }
+   *    },
+   *    sub: "did:ethr:0x413daa771a2fc9c5ae5a66abd144881ef2498c54"
+   *  }
+   *  connect.createVerificationRequest(unsignedClaim).then(jwt => {
+   *    ...
+   *  })
+   *
+   *  @param    {Object}      reqObj                 object with request params
+   *  @param    {Object}      reqObj.unsignedClaim   an object that is an unsigned claim which you want the user to attest
+   *  @param    {String}      reqObj.sub             the DID which the unsigned claim is about
+   *  @param    {String}      [id='signClaimReq']    string to identify request, later used to get response
+   */
+  createVerificationRequest (reqObj, id = 'signClaimReq') {
```

Comment:
How about naming this `.requestVerification()` instead, to make this more consistent with `.attest()` and `.requestDisclosure()`?  This would also more clearly highlight the added responsibility of sending the results from the wrapped `uport-js` functions given the changes proposed in [this comment](https://github.com/uport-project/uport-js/pull/61#issuecomment-415158528).

Reply:
Yes I agree. `requestVerification()` is clearer.